contracts/src/protocol/OmniPortal.sol

```diff
@@ -310,4 +310,13 @@ contract OmniPortal is IOmniPortal, IOmniPortalAdmin, OwnableUpgradeable, OmniPo
 
         emit ValidatorSetAdded(valSetId);
     }
+
+    /**
+     * @notice Query a validator set total power by id
+     * @param valSetId      Validator set id
+     * @return totalPower  Total power of the validator set
+     */
+    function getValidatorSetTotalPower(uint64 valSetId) external view returns (uint64 totalPower) {
+        return validatorSetTotalPower[valSetId];
+    }
```

Comment:
needed to query the portal for "do you know about this validator set", this seemed the easiest.